Katana/SideEffect.swift

```diff
@@ -11,66 +11,141 @@ import Hydra
 
 /**
  Type erasure for `SideEffectContext`
- 
+
  - seeAlso: `SideEffectContext`
  */
 public protocol AnySideEffectContext {
-  
+
   /// Type erased dependencies of the side effect
   var anyDependencies: SideEffectDependencyContainer { get }
-  
+
   /**
    Type erased function that returns the current configuraiton of the state
-   
+
    - returns: the type erased current configuration of the state
    */
   func getAnyState() -> State
-  
+
   /**
    Dispatches a `Dispatchable`. It will esentially call the `anyDispatch` method of the backing store.
-   
+
    - parameter dispatchable: the item to dispatch
    - returns: a promise that is resolved when the store finishes handling the dispatched item
    */
   @discardableResult
   func anyDispatch(_ dispatchable: Dispatchable) -> Promise<Any>
```

Comment:
I feel that we can remove this. We have a method that does the same just below this one.
`func anyDispatch(_ dispatchable: Dispatchable) -> Promise<Any>` is not doing anything different from
`func dispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Any>`